### ISSUE TITLE
Matrix job support

### ISSUE THREAD

**Issue by samrocketman** [samrocketman/jervis#7]
Jervis needs to create matrix jobs from more complicated `.jervis.yml` files which test multiple things.

**Comment by samrocketman** [comment]
Partial support was implemented in 4bd3291a2aec695168a22eff2fd600806a351347.

Still need to support:

```yaml
env:
  global:
    - FOOBAR=foobar
  matrix:
    - FOO=foo BAR=bar
    - FOO=bar BAR=foo
```

**Comment by samrocketman** [comment]
Matrix exclusion has been implemented in e4edfa8352f11644eac4a27aeb57acb7cc468d05.

**Comment by samrocketman** [comment]
Matrix support has been fully implemented today leading up to commit 591f6bb8c3a6314db04f84514ffd9a546c340dd1.
